File: javascript/rotational-cipher/rotational-cipher.js
const LOWER_ALPHABET = 'abcdefghijklmnopqrstuvwxyz'; 
const UPPER_ALPHABET = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ';

export const rotate = (input, key) => 
    [...input].map(letter => translateLetter(letter, key)).join('');

const translateLetter = (letter, key) => {

    if (!letter.match(/[a-zA-Z]/g)) {
        return letter;
    }

    const ALPHABET = letter.match(/[a-z]/g) ? LOWER_ALPHABET : UPPER_ALPHABET;

    return ALPHABET[(ALPHABET.indexOf(letter) + key) % 26];
};

File: python/rotational-cipher/rotational_cipher.py
LOWER_ALPHABET = "abcdefghijklmnopqrstuvwxyz" 
UPPER_ALPHABET = "ABCDEFGHIJKLMNOPQRSTUVWXYZ"

def encode_letter(letter, key):

    if not letter.isalpha():
        return letter

    alphabet = LOWER_ALPHABET if letter.islower() else UPPER_ALPHABET

    position = (alphabet.find(letter) + key) % 26

    return alphabet[position]

def rotate(text, key):

    res = ""
    
    for letter in text:
        res += encode_letter(letter, key)

    return res
